vsintegration/src/FSharp.Editor/Common/LanguageService.fs

```diff
@@ -245,28 +242,38 @@ and
     internal FSharpLanguageService(package : FSharpPackage) =
     inherit AbstractLanguageService<FSharpPackage, FSharpLanguageService>(package)
 
-    let checkerProvider = package.ComponentModel.DefaultExportProvider.GetExport<FSharpCheckerProvider>().Value
     let projectInfoManager = package.ComponentModel.DefaultExportProvider.GetExport<ProjectInfoManager>().Value
 
     let projectDisplayNameOf projectFileName = 
         if String.IsNullOrWhiteSpace projectFileName then projectFileName
         else Path.GetFileNameWithoutExtension projectFileName
 
-    let openedProjects = Queue<IProvideProjectSite>()
+    let singleFileProjects = ConcurrentDictionary<_, AbstractProject>()
 
-    do
-      Events.SolutionEvents.OnAfterOpenProject.Add (fun (args: Events.OpenProjectEventArgs) -> 
-        match args.Hierarchy with
-        | :? IProvideProjectSite as siteProvider ->
-            openedProjects.Enqueue(siteProvider)            
-        | _ -> ())
-        
-    member this.BatchSetupProjects() =    
-        if openedProjects.Count > 0 then    
-            let workspace = package.ComponentModel.GetService<VisualStudioWorkspaceImpl>()
-            for siteProvider in openedProjects do
-                this.SetupProjectFile(siteProvider, workspace)
-            openedProjects.Clear()
+    let tryRemoveSingleFileProject projectId =
+        match singleFileProjects.TryRemove(projectId) with
+        | true, project ->
+            projectInfoManager.RemoveSingleFileProject(projectId)
+            project.Disconnect()       
+        | _ -> ()
+
+    override this.Initialize() =
+        base.Initialize()
+ 
+        this.Workspace.Options <- this.Workspace.Options.WithChangedOption(Completion.CompletionOptions.BlockForCompletionItems, FSharpCommonConstants.FSharpLanguageName, false)
```

Comment:
I moved it to Initialize(), it was needlessly (I think) set each time SetupNewTextView() ran. Options set should be in line with current master.